File: leetcode/509-fibonacci-number/fibonacci-number.py
class Solution:
    def fib(self, N: int) -> int:
        sum = 0
        stack = [N]

        while(len(stack) > 0):
            val = stack.pop()

            if val == 0:
                sum += 0
            elif val == 1:
                sum += 1
            else:
                stack.append(val - 2)
                stack.append(val - 1)

        return sum

File: leetcode/509-fibonacci-number/fibonacci-number.js
/**
 * @param {number} N
 * @return {number}
 */
var fib = function (N) {
  let sum = 0;
  const stack = [];
  stack.push(N);

  while (stack.length > 0) {
    const val = stack.pop();

    if (val === 0) {
      sum += 0;
    } else if (val === 1) {
      sum += 1;
    } else {
      stack.push(val - 2);
      stack.push(val - 1);
    }
  }

  return sum;
};
